module-code/app/securesocial/controllers/LoginApi.scala

```diff
@@ -32,13 +38,21 @@ import securesocial.core.services.SaveMode
 /**
  * A default controller that uses the BasicProfile as the application user type.
  */
-class LoginApi @Inject() (implicit val env: RuntimeEnvironment, val configuration: Configuration, val playEnv: Environment) extends BaseLoginApi
+class LoginApi @Inject() (implicit val env: RuntimeEnvironment, val configuration: Configuration, val playEnv: Environment,
+    langs: Langs,
+    action: DefaultActionBuilder,
+    parsers: PlayBodyParsers,
+    val messagesApi: MessagesApi,
+    val parser: BodyParsers.Default,
+    val fileMimeTypes: FileMimeTypes,
+    val config: ParserConfiguration,
+    val errorHandler: HttpErrorHandler,
+    val materializer: Materializer,
```

Comment:
I'm not so sure, but if you refactor the `SecureSocial`, the constructors of the child classes might become simpler. Can you revisit the following pages to see if you can refactor the classes?
https://www.playframework.com/documentation/2.6.x/Migration26
https://www.playframework.com/documentation/2.6.x/ScalaBodyParsers

Reply:
My idea: absorb implicit decls.  in SecureSocial trait by using field injections.
I will attempt to this idea into code...